src/s3ql/database.py

```diff
@@ -94,6 +94,16 @@ def serialize(self):
         return freeze_basic_mapping(dataclasses.asdict(self))
 
 
+# convert unsigned time_ns (64-bit int) to signed (for db insert/update)
```

Comment:
I do not think this code is correct. Python does not distinguish between signed and unsigned integers. Rather, what you are doing is adding a large constant. I suspect that you are trying to undo the effect of an unsigned number being re-interpreted as a signed number in two-complement notation by C code somewhere. Is that right?

If so, then this is a very fragile way to do this that's prone to misuse. Can't we instead make sure that the number is not misinterpreted in the first case?